hydromt/models/components/grid.py

```diff
@@ -13,371 +11,165 @@
-            Index column should match values in `raster_fn`.
-        variable: str, optional
-            Name of raster_fn dataset variable to use. This is only required when
-            reading datasets with multiple variables.
-            By default None.
-        fill_method : str, optional
-            If specified, fills nodata values in `raster_fn` using fill_nodata method
-            before reclassifying. Available methods are
-            {'linear', 'nearest', 'cubic', 'rio_idw'}.
-        reproject_method: str, optional
-            See rasterio.warp.reproject for existing methods, by default "nearest".
-            Can provide a list corresponding to ``reclass_variables``.
-        mask_name: str, optional
-            Name of mask in self.grid to use for masking raster_fn. By default 'mask'.
-            Use None to disable masking.
-        rename: dict, optional
-            Dictionary to rename variable names in reclass_variables before adding to
-            grid {'name_in_reclass_table': 'name_in_grid'}. By default empty.
-
+        fn : str, optional
+            filename relative to model root, by default 'grid/grid.nc'
         **kwargs : dict
-            Additional keyword arguments to be passed to `get_rasterdataset`
-
-        Returns
-        -------
-        list
-            Names of added model grid layers
-        """  # noqa: E501
-        rename = rename or dict()
-        self.logger.info(
-            f"Preparing grid data by reclassifying the data in {raster_fn} based "
-            f"on {reclass_table_fn}"
-        )
-        # Read raster data and remapping table
-        da = self.data_catalog.get_rasterdataset(
-            raster_fn, geom=self.region, buffer=2, variables=variable, **kwargs
-        )
-        if not isinstance(da, xr.DataArray):
-            raise ValueError(
-                f"raster_fn {raster_fn} should be a single variable. "
-                "Please select one using the 'variable' argument"
+            Additional keyword arguments to be passed to the `write_nc` method.
+        gdal_compliant : bool, optional
+            If True, write grid data in a way that is compatible with GDAL,
+            by default False
+        rename_dims: bool, optional
+            If True and gdal_compliant, rename x_dim and y_dim to standard names
+            depending on the CRS (x/y for projected and lat/lon for geographic).
+        force_sn: bool, optional
+            If True and gdal_compliant, forces the dataset to have
+            South -> North orientation.
+        """
+        if len(self.data) == 0:
+            _exec_nodata_strat(
+                msg="No grid data found, skip writing.",
+                strategy=NoDataStrategy.IGNORE,
+                logger=logger,
+            )
+        else:
+            if not self.root.is_writing_mode():
+                raise IOError("Model opened in read-only mode")
```

Comment:
not exactly sure if this is the right place to check this. We can definitely keep this but we'll also have to check at some point whether files exist and whether we're in overwrite mode. If you did this else where my bad, but I didn't think I saw that.